assets/wizards/dashboard/components/site-statuses/site-status.tsx

```diff
@@ -0,0 +1,147 @@
+ * Dependencies
+ */
+// WordPress
+import { __, _n, sprintf } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import { useState, useEffect } from '@wordpress/element';
+import { Tooltip } from '@wordpress/components';
+// Internal
+import SiteActionModal from './site-status-modal';
+
+const defaultStatusLabels = {
+	idle: '',
+	success: __( 'Connected', 'newspack-plugin' ),
+	pending: __( 'Fetching…', 'newspack-plugin' ),
+	'pending-install': __( 'Installing…', 'newspack-plugin' ),
+	// Error types
+	error: __( 'Disconnected', 'newspack-plugin' ),
+};
+
+const SiteStatus = ( {
+	label = '',
+	canConnect = true,
+	dependencies: dependenciesProp = null,
+	statusLabels,
+	endpoint,
+	then,
+}: SiteStatus ) => {
+	const [ requestStatus, setRequestStatus ] = useState< Statuses >( 'idle' );
+	const [ failedDependencies, setFailedDependencies ] = useState< string[] >( [] );
+	const [ isModalVisible, setIsModalVisible ] = useState( false );
+	const parsedStatusLabels = { ...defaultStatusLabels, ...statusLabels };
+
+	const dependencies = structuredClone( dependenciesProp ) as Dependencies;
+
+	useEffect( () => {
+		makeRequest();
+	}, [] );
+
+	const makeRequest = ( pluginInfo = {} ) => {
+		// When/if a dependency is activated update reference.
+		if ( dependencies && Object.keys( pluginInfo ).length > 0 ) {
+			for ( const [ pluginName ] of Object.entries( pluginInfo ) ) {
+				dependencies[ pluginName ].isActive = true;
+			}
+		}
+		return new Promise( ( resolve, reject ) => {
+			// Dependency check
+			if ( dependencies && Object.keys( dependencies ).length > 0 ) {
+				const failedDeps: string[] = [];
+				for ( const [ dependencyName, dependencyInfo ] of Object.entries( dependencies ) ) {
+					// Don't process active
+					if ( dependencyInfo.isActive ) {
+						continue;
+					}
+					failedDeps.push( dependencyName );
+				}
+				setFailedDependencies( failedDeps );
+				if ( failedDeps.length > 0 ) {
+					setRequestStatus( 'error-dependency' );
+					resolve( false );
+					return;
+				}
+			}
+			// Preflight check
+			if ( ! canConnect ) {
+				setRequestStatus( 'error' );
+				resolve( false );
+				return;
+			}
+			// Pending API request
+			setRequestStatus( 'pending' );
+			apiFetch( {
+				path: endpoint,
+			} )
+				.then( data => {
+					const apiRequest = then( data );
+					setRequestStatus( apiRequest ? 'success' : 'error' );
+					resolve( apiRequest );
+				} )
+				.catch( () => {
+					then( false );
+					setRequestStatus( 'error' );
+					reject();
+				} );
+		} );
+	};
+
+	const classes = `newspack-site-status newspack-site-status__${ requestStatus }`;
+
+	const statusLabel =
+		dependencies && 'error-dependency' === requestStatus ? (
+			<Tooltip
+				text={ sprintf(
+					// translators: %s is a comma separated list of needed dependencies.
+					__( '%s must be installed & activated!' ),
+					failedDependencies.map( dep => dependencies[ dep ].label ).join( ', ' )
+				) }
+			>
+				<button
```

Comment:
I love how if you're missing dependencies, clicking on the status component will pop up the modal to install the missing dependencies. But if all dependencies are installed and the status is still showing "Disconnected", it's not immediately clear how to resolve the status to "Connected". In that case, could we make clicking on the disconnected status redirect to the relevant WP admin page to complete setup?

- Reader Activation: `/wp-admin/admin.php?page=newspack-engagement-wizard#/reader-activation`
- Google Analytics: `/wp-admin/admin.php?page=googlesitekit-splash` if Site Kit isn't connected, or `/wp-admin/admin.php?page=googlesitekit-settings#/connected-services/analytics-4` if Site Kit is connected but Analytics isn't
- GAM: `/wp-admin/admin.php?page=newspack-advertising-wizard#/`

Reply:
Thanks for this feedback, have updated to include config links when status fails.